doi.jq

```diff
@@ -0,0 +1,186 @@
+		"varnish",
+		"vault",
+		"yourls",
+		"znc",
+		"adminer",
+		"aerospike",
+		"amazoncorretto", # direct children: jetty maven tomcat
+		"backdrop",
+		"buildpack-deps", # direct children: erlang gcc golang haskell haxe influxdb kapacitor node openjdk perl pypy python rakudo-star ruby rust telegraf
+		"composer", # direct children: drupal
+		"couchbase",
+		"dart",
+		"eclipse-temurin", # direct children: cassandra clojure flink gradle groovy jetty jruby lightstreamer liquibase maven neo4j orientdb solr sonarqube spark storm tomcat tomee unit zookeeper
+		"erlang", # direct children: elixir
+		"flink",
+		"friendica",
+		"gazebo",
+		"gradle",
+		"groovy",
+		"haskell",
+		"haxe",
+		"ibm-semeru-runtimes", # direct children: maven open-liberty tomee websphere-liberty
+		"ibmjava", # direct children: maven websphere-liberty
+		"jetty", # direct children: geonetwork
+		"joomla",
+		"jruby",
+		"kong",
+		"lightstreamer",
+		"liquibase",
+		"matomo",
+		"mediawiki",
+		"monica",
+		"neurodebian",
+		"nextcloud",
+		"node", # direct children: express-gateway ghost mongo-express unit
+		"odoo",
+		"open-liberty",
+		"orientdb",
+		"python", # direct children: hylang plone satosa unit
+		"rakudo-star",
+		"ros",
+		"sapmachine", # direct children: maven
+		"satosa",
+		"silverpeas",
+		"spark",
+		"storm",
+		"swift",
+		"tomcat", # direct children: convertigo geonetwork xwiki
+		"tomee",
+		"websphere-liberty",
+		"zookeeper",
+		"clojure",
+		"convertigo",
+		"geonetwork",
+		"maven",
+		"mongo-express",
+		"plone",
+
+		empty
+	]
+;
+
+# a helper for "build_should_sbom"
+def _sbom_subset:
+	[
+		# only repositories we have explicitly verified
+		"bash",
+		"buildpack-deps",
+		"busybox",
+		"caddy",
+		"cassandra",
+		"chronograf",
+		"couchdb",
+		"drupal",
+		"eclipse-temurin",
+		"elasticsearch",
+		"fluentd",
+		"gcc",
+		"ghost",
+		"haproxy",
+		"httpd",
+		"hylang",
+		"influxdb",
+		"julia",
+		"kapacitor",
+		"kibana",
+		"logstash",
+		"mariadb",
+		"memcached",
+		"mongo",
+		"mysql",
+		"neo4j",
+		"nginx",
+		"openjdk",
+		"perl",
+		"php",
+		"postgres",
+		"pypy",
+		"python",
+		"rabbitmq",
+		"redis",
+		"registry",
+		"ruby",
+		"rust",
+		"solr",
+		"sonarqube",
+		"telegraf",
+		"tomcat",
+		"wordpress",
+		"xwiki",
+		empty
+	]
+;
+
+# input: "build" object (with "buildId" top level key)
+# output: boolean
+def build_should_sbom:
+	.source.arches[.build.arch].tags
+	| map(split(":")[0])
+	| unique
+	| _sbom_subset as $subset
+	| any(.[];
+		. as $i
+		| $subset
+		| index($i)
+	)
+;
+
+# input: "build" object (with "buildId" top level key)
+# output: boolean
+def build_should_sign:
+	.build.arch == "amd64" and (
+		.source.arches[.build.arch].tags
+		| map(split(":")[0])
+		| unique
+		| index("notary")
+	)
+;
```

Comment:
Expected usage of this (https://github.com/docker-library/meta/blob/174af2e0f74540b250f365d4cacfe5fce4b45990/.github/workflows/build.yml#L110) would look something like:

```patch
diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index fceac66..0a2dd7d 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -107,7 +107,7 @@ jobs:
           mkdir build
 
           # TODO signing prototype -- starting very small
-          shouldSign="$(jq <<<"$json" '.build.arch == "amd64" and any(.source.arches[.build.arch].tags[]; startswith("notary:"))')"
+          shouldSign="$(jq <<<"$json" -L.scripts 'include "doi"; build_should_sign')"
           [ "$shouldSign" = 'true' ] || [ "$shouldSign" = 'false' ] || exit 1
           echo "shouldSign=$shouldSign" >> "$GITHUB_ENV"
```